### DISCUSSION TITLE
CloudQuery: High-performance data integration framework

### DISCUSSION THREAD

**Discussion by amotl** [crate/crate-clients-tools#299] [Ideas]
## About
> CloudQuery is an open-source, high-performance data integration framework built for developers.
>
> CloudQuery extracts, transforms, and loads configuration from cloud APIs to variety of supported destinations such as databases, data lakes, or streaming platforms for further analysis.

## References
- https://github.com/cloudquery/cloudquery
- https://www.cloudquery.io/
- https://github.com/cloudquery/cloudquery/tree/main/plugins/destination/postgresql

/cc @hlcianfagna

**Comment by wierdvanderhaar** [comment]
We had a conversation with the CEO and an engineer from CQ. They are probably able to solve the issue that cq columns are created starting with an underscore which doesn't work with CrateDB. See this issue at their end. 
https://github.com/cloudquery/cloudquery/issues/17043

**Comment by amotl** [comment]
That has converged fast. Thank you so much, @hermanschaaf.

- https://github.com/cloudquery/cloudquery/pull/17088

**Comment by amotl** [comment]
Hi again. CrateDB 6.2 removed the restriction to store columns with leading underscore characters. Only `_id` is not permitted (amongst other reserved ones), just to name the most prominent counter-example.

Do you have an idea how to loop in that change to the CrateDB adapter, @hermanschaaf?

**Comment by hermanschaaf** [comment]
Hi @amotl, I'm no longer working on this but it sounds like you would mostly be able to remove some special cases now, since CloudQuery uses columns like `_cq_id` and `_cq_sync_time`, and I guess we had to special-case it when we implemented the CrateDB adapter. 

I think there's a good chance Claude / Codex would be able to handle this one-shot :)

**Reply by amotl** [reply]
Hi Herman. This sounds good! Do you want us to submit a corresponding patch, or can you loop it in on your own?
